Automattic/jetpack
# Gutenberg: Add Goodreads Block

Fixes Automattic/wp-calypso#74494
Fixes #7920 
## Proposed changes:


Adds a block for Goodreads so that users can update to newer themes - it's currently only available as a legacy widget. This block is similar to the existing widget in that it relies on the default Goodreads styles, but I've added a little bit more customisation which Goodreads offers that was never included in the legacy widget. This can be seen in the Settings panel, and it just controls which books display.

Goodreads also offers a Grid widget that was never used in Jetpack. I feel like Gutenberg is perfect at handling multiple styles, so I've added it with this block.

| Grid Widget                                      | Default Widget                                     |
| -------------------------------------------- | -------------------------------------------- |
| ![Screenshot 1](https://github.com/Automattic/jetpack/assets/43215253/69c77d3e-90f2-4638-9735-64db0b774aec) | ![Screenshot 2](https://github.com/Automattic/jetpack/assets/43215253/6e297023-b2ef-4637-9b13-9fa2f7706528) |

I'm marking this as a draft because @supernovia mentioned in the original issue that it'd be good to support authors as well as readers. Unfortunately, I don't have an Authors Goodreads account, so I can't yet see how to implement those widgets. I'll do a bit more digging on that, but would be very grateful for any early feedback - it's been a while since I've tried my hand at making a block!

## Does this pull request change what data or activity we track or use?


No changes. Regarding the above section, I don't believe that I'm able to test my code on WordPress.com, and I'm not familiar with e2e tests for Jetpack.

## Testing instructions:






- You should be able to create a Goodreads block using a profile URL - so for example: https://www.goodreads.com/user/show/1176283-matt-mullenweg
- Verify that it loads properly and that each of the settings in the sidebar works 
- Hopefully nothing is too out of place! 

Notably, you should also be able to test this with an Author URL. The Goodreads Widget directs users to documentation on how to acquire their user ID because it's different for authors; with this block, it's much simpler and only requires your profile URL. Example author profile: https://www.goodreads.com/author/show/1406384
